### PR TITLE
Started Backend models/services and implementing CRUD (without roles) for Stock entity using external API

### PR DESCRIPTION
**Pull request by AntoineAssal**
Before accepting and merging please make sure everything works as expected.

**Requirements**:
- Postman, thunderbird or any HTTP client (API keys on discord)
- Access to the db. I called it `trading`, can change in app properties. (Table create statements on discord)

## 1. Authentication Test (no user)
 
POST `/login`
Content-Type: `application/json`

Include credentials as JSON
```json
{
  "username": "testuser",
  "password": "testpassword"
}
```
Your return (since user doesn't exist) should be:
```json
{
    "message": "Unknown user"
}
```

## 2. Register Test

POST `/register`
Content-Type: `application/json`

Include credentials as JSON
```json
{
  "username": "User",
  "password": "secure_password",
  "email": "user@example.com"
}
```
Your return should be (a random token):
```json
{
    "id": 1,
    "username": User,
    "email": user@example.com,
    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOm51bGwsImV4cCI6MTcxMjk3MDE0NSwiaWF0IjoxNzEyOTY2NTQ1fQ.BIVSQKt96VGHrR_BEjQfhB8EDivPZ0M8hM8g8_4DpVo",
    "role": null
}
```

## 3.Authentication Test (Exists but wrong password)
 
POST `/login`
Content-Type: `application/json`

Include credentials as JSON
```json
{
  "username": "testuser",
  "password": "testwrong"
}
```
Your return should be:
```json
{
    "message": "Invalid password"
}
```

## 4.Authentication Test (All good)
 
POST `/login`
Content-Type: `application/json`

Include credentials as JSON:
```json
{
  "username": "Usera",
  "password": "secure_password",
  "email": "user@example.com"
}
```
Your return should be:
```json
{
    "id": 3,
    "username": "Usera",
    "email": "user@example.com",
    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJVc2VyYSIsImV4cCI6MTcxMzEwNDk0MywiaWF0IjoxNzEzMTAxMzQzfQ.eRHSgjtQCrPAuUSqIPCxlC2RUPnXmbLAQAlEHtxgSIg",
    "role": null
}
```

## 4. Stock Creation Test (role access still WIP, anyone can create at the moment)
 
POST `/admin/stocks?symbol=<ticker>`
Query Parameters `ticker` for example, `AAPL`

Your return (in Intellij Console) should be:
```
DTO fetched: StockApiDto(id=b0b6dd9d-8b9b-48a9-ba46-b9d54906e415, 
classType=us_equity, 
exchange=NASDAQ, 
symbol=AAPL,
name=Apple Inc. Common Stock, 
status=active, 
tradable=true,
marginable=true,
maintenanceMarginRequirement=0,
shortable=true,
easyToBorrow=false,
fractionable=true,
attributes=[fractional_eh_enabled, options_enabled])


Mapped Stock: Stock(id=null, symbol=AAPL, name=Apple Inc. Common Stock, exchange=NASDAQ, lastPrice=null)
```

### CODE REVIEW

File: backend/src/main/java/wileyt3/backend/controller/AdminController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        Stock stock = stockDataService.fetchStockData(symbol);
+        if (stock != null) {
+            stock = stockDataService.saveStock(stock);  // Save the stock to the database
+            return ResponseEntity.ok(stock);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @PostMapping("/admin/cryptos")
```
Comment (AntoineAssal) [inline backend/src/main/java/wileyt3/backend/controller/AdminController.java:34]: comment out the crypto part (u will get missing import)

File: backend/src/main/java/wileyt3/backend/controller/AdminController.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import wileyt3.backend.entity.Crypto;
+import wileyt3.backend.service.StockDataService;
+import wileyt3.backend.service.CryptoDataService;
+
+@RestController
+@RequiredArgsConstructor
+public class AdminController {
+
+    private final StockDataService stockDataService;
+    private final CryptoDataService cryptoDataService;
```
Comment (AntoineAssal) [inline backend/src/main/java/wileyt3/backend/controller/AdminController.java:20]: comment out

**Review by houssein31 — APPROVED** [review]
Looks good

**Review by FrederickMunro — APPROVED** [review]
All these routes are running nicely, well done.

**Pull request merged**